.github/workflows/notebook-test.yml

```diff
@@ -70,4 +73,10 @@ jobs:
           key: ${{ hashFiles('scripts/nb-tester/requirements.txt') }}
 
       - name: Run tox
-        run: tox -- ${{ steps.all-changed-files.outputs.all_changed_files }}
+        run: tox -- ${{ steps.all-changed-files.outputs.all_changed_files }} --write
+
+      - name: Upload executed notebooks
+        uses: actions/upload-artifact@v4
+        with:
+          name: Executed notebooks
+          path: ${{ steps.all-changed-files.outputs.all_changed_files }}
```

Comment:
Instead, what we did in Qiskit CI is write the notebooks to a dedicated folder `executed_tutorials`. Rather than mutating the existing files.

This currently will upload more files than relevant. If you make this change, I think you can revert your changes to defining `all_changed_files`.

If you still want the script by default to modify in-place, then maye add an argument like `--output-dir`.

Reply:
Good call, the action wasn't handling multiple notebooks correctly. It should be fixed now; see the artifact for [run#7298447436](https://github.com/Qiskit/documentation/actions/runs/7298447436).